app/src/hooks/useNotification.tsx

```diff
@@ -0,0 +1,184 @@
+};
+
+export const NotificationProvider: React.FC = ({ children }) => {
+  const notification = useProvideNotification();
+  return (
+    <NotificationContext.Provider value={notification}>
+      {children}
+    </NotificationContext.Provider>
+  );
+};
+
+const EXPO_NOTIFICATION_URL = 'https://exp.host/--/api/v2/push/send';
+
+Notifications.setNotificationHandler({
+  handleNotification: async () => ({
+    shouldShowAlert: true,
+    shouldPlaySound: false,
+    shouldSetBadge: false,
+  }),
+});
+
+const useProvideNotification = () => {
+  const [token, setToken] = useState<string | undefined | null>(null);
+  const [isLoading, setLoading] = useState(true);
+  const notificationListener = useRef<Subscription>();
+  const responseListener = useRef<Subscription>();
+  const [
+    notification,
+    setNotification,
+  ] = useState<Notifications.Notification>();
+
+  const sendPushNotification = async (
+    title: string,
+    body: string,
+    data: NotificationData
+  ) => {
+    const message = {
+      to: token, //TODO: should the token be passed dynamically? or can it be just passed from state inside this function?
+      sound: 'default',
+      title,
+      body,
+      data,
+    };
+
+    await fetch(EXPO_NOTIFICATION_URL, {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Accept-encoding': 'gzip, deflate',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(message),
+    });
+  };
+
+  const registerForPushNotificationsAsync = async () => {
+    if (Constants.isDevice) {
+      const { status: existingStatus } = await Permissions.getAsync(
+        Permissions.NOTIFICATIONS
+      );
+      let finalStatus = existingStatus;
+      if (existingStatus !== 'granted') {
+        const { status } = await Permissions.askAsync(
+          Permissions.NOTIFICATIONS
+        );
+        finalStatus = status;
+      }
+
+      if (finalStatus !== 'granted') {
+        alert('Failed to get push token for push notification!');
+        return;
+      }
+      const token = (await Notifications.getExpoPushTokenAsync()).data;
+      return token;
+    } else {
+      alert('Must use physical device for Push Notifications');
+    }
+
+    if (Platform.OS === 'android') {
+      await Notifications.setNotificationChannelAsync('default', {
+        name: 'default',
+        importance: Notifications.AndroidImportance.MAX,
+        vibrationPattern: [0, 250, 250, 250],
+        lightColor: '#FF231F7C',
+      });
+    }
+
+    return null;
+  };
+
+  useEffect(() => {
+    const initNotificationUsage = async () => {
+      const storedNotificationToken = await getStoredNotificationToken();
+      setToken(storedNotificationToken);
+      setLoading(false);
+
+      // This listener is fired whenever a notification is received while the app is foregrounded
+      notificationListener.current = Notifications.addNotificationReceivedListener(

```

Comment:
`addNotificationReceivedListener` zwraca subscription, którego potem można użyć żeby usunąć listener. (https://docs.expo.io/versions/v40.0.0/sdk/notifications/#addnotificationreceivedlistenerlistener-event-notification--void-void)

```suggestion
      const subscription = Notifications.addNotificationReceivedListener(
```